common/autoware_universe_utils/include/autoware/universe_utils/ros/static_transform_buffer.hpp

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE__UNIVERSE_UTILS__ROS__STATIC_TRANSFORM_BUFFER_HPP_
+#define AUTOWARE__UNIVERSE_UTILS__ROS__STATIC_TRANSFORM_BUFFER_HPP_
+
+#include "autoware/universe_utils/ros/transform_listener.hpp"
+
+#include <eigen3/Eigen/Core>
+#include <pcl_ros/transforms.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/point_cloud2_iterator.hpp>
+
+#include <pcl_conversions/pcl_conversions.h>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+namespace std
+{
+template <>
+struct hash<std::pair<std::string, std::string>>
+{
+  size_t operator()(const std::pair<std::string, std::string> & p) const
+  {
+    size_t h1 = std::hash<std::string>{}(p.first);
+    size_t h2 = std::hash<std::string>{}(p.second);
+    return h1 ^ (h2 << 1u);
+  }
+};
+}  // namespace std
+
+namespace autoware::universe_utils
+{
+using std::chrono_literals::operator""ms;
+using Key = std::pair<std::string, std::string>;
+struct PairEqual
+{
+  bool operator()(const Key & p1, const Key & p2) const
+  {
+    return p1.first == p2.first && p1.second == p2.second;
+  }
+};
+using TFMap = std::unordered_map<Key, Eigen::Matrix4f, std::hash<Key>, PairEqual>;
+
+class StaticTransformBuffer
+{
+public:
+  StaticTransformBuffer() = default;
+
+  /**
+   * @brief Retrieves a transform between two static frames.
+   *
+   * This function attempts to retrieve a transform between the target frame and the source frame.
+   * If success, the transform matrix is set to the output parameter and the function returns true.
+   * Otherwise, transform matrix is set to identity and the function returns false. Transform
+   * Listener is destroyed after function call.
+   *
+   * @param node A pointer to the ROS node.
+   * @param target_frame The target frame.
+   * @param source_frame The source frame.
+   * @param eigen_transform The output Eigen transform matrix. Is set to identity if the transform
+   * is not found.
+   * @return True if the transform was successfully retrieved, false otherwise.
+   */
+  bool getTransform(
```

Comment:
@amadeuszsz 
For the next time, unless we are worrying about heavy algorithms, instead of modifying a reference and returning a boolean, std::optional could be a good alternative

Reply:
Filter class already follow such an approach ([modifying a reference](https://github.com/autowarefoundation/autoware.universe/blob/d4f365da2f2e6403381ce5c64e60d3820428128a/sensing/autoware_pointcloud_preprocessor/src/filter.cpp#L307)), so I didn't find any reason to change this. If there are performance, principle or readability reason we can address this point for whole package during code refactor.

Reply:
Thanks for the clarification !